specs/merkle-tree.md

```diff
@@ -0,0 +1,158 @@
+# Sphinx Merkle Tree
+Sphinx uses a custom Merkle tree data structure to allow teams to approve arbitrarily large deployments across an arbitrary number of networks via off-chain signatures, and then allow those transactions to be trustlessly executed via their Gnosis Safe by a third party relayer. This document outlines the architecture and content of the Sphinx Merkle tree.
+
+**Vocabulary notes**:
```

Comment:
I think we should add a vocabulary term for the word _index_. We should make it clear that this refers to a field in the Merkle leaf, and _not_ the Merkle leaf's position in the Merkle tree. I think it'd be appropriate to add a link or a footnote explaining why we do this.